apps/dokploy/components/dashboard/docker/logs/docker-logs-id.tsx

```diff
@@ -1,110 +1,158 @@
-				wsRef.current.close();
-			}
-			wsRef.current = null;
-		}
-		const termi = new Terminal({
-			cursorBlink: true,
-			cols: 80,
-			rows: 30,
-			lineHeight: 1.25,
-			fontWeight: 400,
-			fontSize: 14,
-			fontFamily:
-				'ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace',
-
-			convertEol: true,
-			theme: {
-				cursor: "transparent",
-				background: "rgba(0, 0, 0, 0)",
-			},
-		});
+		container.innerHTML = "";
+		wsRef.current?.close();
+
+		const terminal = new Terminal(TERMINAL_CONFIG);
+		const fitAddon = new FitAddon();
+		terminal.loadAddon(fitAddon);
+		terminal.open(container);
+		fitAddon.fit();
+		terminal.focus();
+		setTerm(terminal);
 
 		const protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
+		const wsUrl = `${protocol}//${window.location.host}/docker-container-logs?containerId=${containerId}&tail=${lines}${serverId ? `&serverId=${serverId}` : ""}&timeFilter=${timeFilter}`;
 
-		const wsUrl = `${protocol}//${window.location.host}/docker-container-logs?containerId=${containerId}&tail=${lines}${serverId ? `&serverId=${serverId}` : ""}`;
 		const ws = new WebSocket(wsUrl);
 		wsRef.current = ws;
-		const fitAddon = new FitAddon();
-		termi.loadAddon(fitAddon);
-		// @ts-ignore
-		termi.open(container);
-		fitAddon.fit();
-		termi.focus();
-		setTerm(termi);
-
-		ws.onerror = (error) => {
-			console.error("WebSocket error: ", error);
-		};
 
-		ws.onmessage = (e) => {
-			termi.write(e.data);
+		ws.onmessage = ({ data }) => {
+			const formattedLog = formatLogLine(data);
+			if (searchTerm.length <= 1 || data.includes(searchTerm)) {
+				terminal.write(`${formattedLog}\r\n`);
+			}
 		};
 
 		ws.onclose = (e) => {
-			console.log(e.reason);
-
-			termi.write(`Connection closed!\nReason: ${e.reason}\n`);
+			terminal.write(`Connection closed!\nReason: ${e.reason}\n`);
 			wsRef.current = null;
 		};
-		return () => {
-			if (wsRef.current?.readyState === WebSocket.OPEN) {
-				ws.close();
-				wsRef.current = null;
-			}
-		};
-	}, [lines, containerId]);
+
+		return () => ws.close();
+	}, [lines, containerId, timeFilter, searchTerm, id, serverId]);
 
 	useEffect(() => {
 		term?.clear();
 	}, [lines, term]);
 
 	return (
 		<div className="flex flex-col gap-4">
-			<div className="flex flex-col gap-2">
-				<Label>
-					<span>Number of lines to show</span>
-				</Label>
-				<Input
-					type="text"
-					placeholder="Number of lines to show (Defaults to 35)"
-					value={lines}
-					onChange={(e) => {
-						setLines(Number(e.target.value) || 1);
-					}}
-				/>
+			<div className="flex items-center gap-2 bg-background/95 rounded-md">
+				<div className="flex flex-col gap-1">
+					<Label htmlFor="time-filter">Lines & Time Range</Label>
+					<Select
+						value={timeFilter}
+						onValueChange={(value: TimeFilter) => {
+							setTimeFilter(value);
+							if (value.endsWith("lines")) {
+								setLines(Number.parseInt(value));
```

Comment:
Would setting lines clear the time filter and vice-versa (it should)? I don't see that logic here.